agreement/models/agreement.py

```diff
@@ -10,7 +10,7 @@ class Agreement(models.Model):
     _description = 'Agreement'
     _inherit = ["mail.thread", "mail.activity.mixin"]
 
-    code = fields.Char(required=True, copy=False)
+    code = fields.Char(required=True)
```

Comment:
why did you do that ? There is a unicity constrain...

Reply:
If field is required you can not duplicate record. Maybe it would be necessary to create a sequence..

Reply:
Another option could be to add "(Copy)" to the code:
https://github.com/odoo/odoo/blob/c5a8051a2c840b2be545eee2f173261a908de737/addons/account/models/account.py#L253
:sweat_smile: